### PR TITLE
MAINT Use class `LearningCurveDisplay`

### PR DESCRIPTION
**Pull request by ArturoAmorQ**
[LearningCurveDisplay](https://scikit-learn.org/dev/modules/generated/sklearn.model_selection.LearningCurveDisplay.html) is available since 1.2. This PR makes use of this new class for easing the plotting a little bit.

### CODE REVIEW

File: python_scripts/cross_validation_learning_curve.py [hunk 1]
```diff
@@ -74,56 +74,35 @@
 # Now, we are all set to carry out the experiment.
 
 # %%
-from sklearn.model_selection import learning_curve
+from sklearn.model_selection import LearningCurveDisplay
+import matplotlib.pyplot as plt
```
Comment (glemaitre) [inline python_scripts/cross_validation_learning_curve.py:78]:
```suggestion
from sklearn.model_selection import LearningCurveDisplay
```

File: python_scripts/cross_validation_learning_curve.py [hunk 2]
```diff
@@ -74,56 +74,35 @@
...
 
-results = learning_curve(
+LearningCurveDisplay.from_estimator(
```
Comment (glemaitre) [inline python_scripts/cross_validation_learning_curve.py:80]:
```suggestion
display = LearningCurveDisplay.from_estimator(
```

File: python_scripts/cross_validation_learning_curve.py [hunk 3]
```diff
@@ -74,56 +74,35 @@
...
-    test_errors.mean(axis=1),
-    yerr=test_errors.std(axis=1),
-    label="Testing error",
-)
-plt.legend()
-
-plt.xscale("log")
-plt.xlabel("Number of samples in the training set")
-plt.ylabel("Mean absolute error (k$)")
 _ = plt.title("Learning curve for decision tree")
```
Comment (glemaitre) [inline python_scripts/cross_validation_learning_curve.py:94]:
```suggestion
_ = display.ax_.set_title("Learning curve for decision tree")
```

**Review by glemaitre — COMMENTED** [review]
nitpicks

**Review by ogrisel — APPROVED** [review]

**Pull request merged**